primitives/src/communities.rs

```diff
@@ -19,9 +43,38 @@ pub struct Location {
 }
 
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Default, RuntimeDebug)]
-pub struct CommunityPropertiesType {
-    pub name_utf8: Vec<u8>,
-    pub demurrage_per_block: Demurrage,
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommunityMetadata {
+    /// utf8 encoded name
+    pub name: PalletString,
+    /// utf8 encoded abbreviation of the name
+    pub symbol: PalletString,
+    /// multi-resolution resource for the community icon
+    pub icons: Vec<Favicon>,
```

Comment:
I'm undecided if we should really bloat the chain with so many ipfs cid's. we could link to a single json on ipfs defining the favicons

Reply:
yeah, let's do a single cid. there is no added value in having this all onchain. immutability is given anyway

Reply:
> For future PR's: Please try keeping them small and make one PR per change to be reviewed. Here I had to review test improvements and cosmetics along with the actual task, which makes it confusing

Yes sorry about that. I agree and I should have taken the extra effort. It kind of emerged organically, as I had to use the `register_test_currency` in a new place and thought that instead of copy-pasting it to yet another location I solve the issue. But I will be extra strict with myself the next time. :)